themes/ddbasic/template.php

```diff
@@ -805,28 +538,22 @@ function ddbasic_remove_default_link_classes($classes) {
 }
 
 /**
- * Allows us to add script plugins to the theme via theme settings.
- *
- * Ex. add a javascript depending on the settings in the theme.
+ * Implements hook_js_alter().
  */
-function ddbasic_load_plugins() {
-  $theme_path = drupal_get_path('theme', 'ddbasic');
-
-  // If sticky menus is enabled in the theme load it.
-  if (theme_get_setting('main_menu_sticky')) {
-
-    // Add variable to js so we can check if it is set.
-    drupal_add_js(array('ddbasic' => array('main_menu_sticky' => theme_get_setting('main_menu_sticky'))), 'setting');
+function ddbasic_js_alter(&$javascript) {
+  // Set the ding_popup.js to the popup-hijack.js instead.
+  $ding_popup = drupal_get_path('module', 'ding_popup') . '/ding_popup.js';
+  if (isset($javascript[$ding_popup])) {
+    $javascript[$ding_popup]['data'] = drupal_get_path('theme', 'ddbasic') . '/scripts/popup-hijack.js';
   }
 
-  // If equalize is enabled in the theme load it.
-  if (theme_get_setting('load_equalize')) {
-
-    // Add the script.
-    drupal_add_js($theme_path . '/scripts/equalize.min.js');
-
-    // Add variable to js so we can check if it is set.
-    drupal_add_js(array('ddbasic' => array('load_equalize' => theme_get_setting('load_equalize'))), 'setting');
+  // Remove the opening_hours files, so they dont't cause a JavaScript error
+  // when outputting the theme specific opening_hours template.
+  $opening_hours_path = drupal_get_path('module', 'opening_hours');
+  foreach ($javascript as $key => $value) {
+    if (strpos($key, $opening_hours_path) !== FALSE) {
+      unset($javascript[$key]);
+    }
```

Comment:
The fact that we have to remove JS like this in our theme is not a good sign. If it conficts with what is going on in ding_ddbasic_opening_hours then it should be moved there.

However I would much prefer a solution where we do not have to resort to such hackery.